src/YQPkgVersionsView.cc

```diff
@@ -522,15 +508,13 @@ YQPkgVersionsView::isMixedMultiVersion( ZyppSel selectable )
 
     bool multiversion = it->multiversionInstall();
 
-    while ( it != selectable->availableEnd() )
+    for ( ; it != selectable->availableEnd(); ++it )
```

Comment:
Not only no, but **hell, no**. This makes everybody reading the code scratch their head.

What is the purpose of eradicating all `while` loops with an iterator? `for` loops have their place, but there is no need to replace every `while` loop with a `for` loop.

Reply:
I removed the changes you didn't like.

Iterators and for loops belong together, they are simple to read. If I could, I turned them into range-base for loops.

Reply:
Some things are natural `for` loops, and some things are not; that's why `while` loops exist.

`for` loops are useful for iterating over a numeric range, or, with modern C++ that supports it, over a container. Technically, each C++ `for` loop can be replaced with a `while` loop and the other way round; but in some cases it just looks weird and unnatural, and in those cases it makes the code harder to understand; such as here.

Reply:
To some extend it is a matter of habit and personal taste. I am glad that we can discuss in a constructive way and we find a consensus in most of the cases.